tests/restricted_rooms_test.go

```diff
@@ -73,88 +71,98 @@ func setupRestrictedRoom(t *testing.T, deployment *docker.Deployment) (*client.C
 func checkRestrictedRoom(t *testing.T, alice *client.CSAPI, bob *client.CSAPI, allowed_room string, room string) {
 	t.Helper()
 
-	failJoinRoom(t, bob, room, "hs1", 403)
-
-	// Join the allowed room, attempt to join the room again, which now should succeed.
-	bob.JoinRoom(t, allowed_room, []string{"hs1"})
-	bob.JoinRoom(t, room, []string{"hs1"})
+	t.Run("JoinShouldFailInitially", func(t *testing.T) {
+		failJoinRoom(t, bob, room, "hs1")
+	})
 
-	// Joining the same room again should work fine (e.g. to change your display name).
-	bob.SendEventSynced(
-		t,
-		room,
-		b.Event{
-			Type:     "m.room.member",
-			Sender:   bob.UserID,
-			StateKey: &bob.UserID,
-			Content: map[string]interface{}{
-				"membership":  "join",
-				"displayname": "Bobby",
-				// This should be ignored since this is a join -> join transition.
-				"join_authorised_via_users_server": "unused",
+	t.Run("JoinShouldSucceedWhenJoinedToAllowedRoom", func(t *testing.T) {
```

Comment:
```suggestion
	t.Run("Join should succeed when joined to allowed room", func(t *testing.T) {
```

We don't need to camel case here.